src/Robo/Plugin/Commands/DevelopmentModeCommands.php

```diff
@@ -0,0 +1,416 @@
+        $this->devServicesPath = "$this->drupalRoot/sites/fe.development.services.yml";
+    }
+
+    /**
+     * Refreshes a development environment.
+     *
+     * Completely refreshes a development environment including running 'composer install', starting Lando, downloading
+     * a database dump, importing it, running 'drush deploy', disabling front-end caches, and providing a login link.
+     *
+     * @param string $siteName
+     *   The Drupal site name.
+     *
+     * @aliases magic
+     *
+     * @return \Robo\Result
+     *   The result of the set of tasks.
+     */
+    public function devRefresh($siteName = 'default'): Result
+    {
+        $this->io()->title('development environment refresh. 🦄✨');
+        $result = $this->taskComposerInstall()->run();
+        $result = $this->taskExec('lando')->arg('start')->run();
+        // There isn't a great way to call a command in one class from another.
+        // https://github.com/consolidation/Robo/issues/743
+        // For now, it seems like calling robo from within robo works.
+        $result = $this->taskExec("composer robo theme:build $siteName")
+            ->run();
+        $result = $this->frontendDevEnable($siteName, ['yes' => true]);
+        $result = $this->databaseRefreshLando($siteName);
+        $result = $this->drupalLoginLink($siteName);
+        return $result;
+    }
+
+    /**
+     * Download the latest database dump for the site.
+     *
+     * @param string $siteName
+     *   The site name.
+     *
+     * @return string
+     *   The path of the last downloaded database.
+     *
+     * @throws \Robo\Exception\TaskException
+     */
+    public function databaseDownload($siteName = 'default')
+    {
+        $this->io()->title('database download.');
+
+        $awsConfigDirPath = getenv('HOME') . '/.aws';
+        $awsConfigFilePath = "$awsConfigDirPath/credentials";
+        if (!is_dir($awsConfigDirPath) || !file_exists($awsConfigFilePath)) {
+            $result = $this->configureAwsCredentials($awsConfigDirPath, $awsConfigFilePath);
+            if ($result->wasCancelled()) {
+                return Result::cancelled();
+            }
+        }
+
+        if (!$bucket = $this->getConfig('database_s3_bucket', $siteName)) {
+            throw new TaskException($this, "database_s3_bucket value not set for '$siteName'.");
+        }
+
+        $s3 = new S3Client();
+        $objects = $s3->listObjectsV2(['Bucket' => $bucket]);
+        $objects = iterator_to_array($objects);
+        // Ensure objects are sorted by last modified date.
+        usort($objects, fn($a, $b) => $a->getLastModified()->getTimestamp() <=> $b->getLastModified()->getTimestamp());
+        $latestDatabaseDump = array_pop($objects);
+        $dbFilename = $latestDatabaseDump->getKey();
+
+        if (file_exists($dbFilename)) {
+            $this->say("Skipping download. Latest database dump file exists >>> $dbFilename");
+        } else {
```

Comment:
```suggestion
        }
        else {
```
Should we enable code style checks on this repo?

Reply:
I see this elsewhere too. Is that intentional?

Reply:
[Code style checks are enabled](https://github.com/ChromaticHQ/chq-robo/blob/c91e5a9021499f8b284444add97303880914ff4c/robo.yml#L12-L13). We are following [PSR-12](https://www.php-fig.org/psr/psr-12/). I don't really have a strong feeling about this but it seemed weird to follow Drupal coding standards in a non-Drupal repo.

For example, [Drush](https://github.com/drush-ops/drush/blob/10.x/CONTRIBUTING.md#coding-style) follows [PSR-2](https://www.php-fig.org/psr/psr-12/) which has been deprecated in favor of PSR-12

Reply:
Interesting. I guess I'm willing to give it a try, but here are my two arguments against it FWIW:

- We are pretty much using this exclusively with Drupal, so it's very much Drupal related.
- It really throws me off when reviewing code. How was it to write it in this format?

Reply:
> We are pretty much using this exclusively with Drupal, so it's very much Drupal related.

I mean, can it get more "Drupal-related" than Drush? They aren't using Drupal coding standards.

> It really throws me off when reviewing code. How was it to write it in this format?

It was fine. When I fell into incorrect habits, [PHPCS would clean it up](https://github.com/ChromaticHQ/chq-robo/blob/c91e5a9021499f8b284444add97303880914ff4c/src/Robo/Plugin/Commands/CICommands.php#L102-L113) for me!

Reply:
> I mean, can it get more "Drupal-related" than Drush? They aren't using Drupal coding standards.

I mean ya haha. I guess I would question this choice too, but 🤷🏼 ?

👍🏼